python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py

```diff
@@ -38,15 +40,172 @@
     )
 
 
-class RemoteAssetGraph(BaseAssetGraph):
+class RemoteAssetNode(BaseAssetNode):
     def __init__(
         self,
-        asset_nodes_by_key: Mapping[AssetKey, "ExternalAssetNode"],
+        key: AssetKey,
+        repo_node_pairs: Sequence[Tuple[RepositoryHandle, "ExternalAssetNode"]],
+        check_keys: AbstractSet[AssetCheckKey],
+        execution_set_keys: AbstractSet[AssetKeyOrCheckKey],
+    ):
+        self.key = key
+        self._repo_node_pairs = repo_node_pairs
+        self._external_asset_nodes = [node for _, node in repo_node_pairs]
+        self._check_keys = check_keys
+        self._execution_set_keys = execution_set_keys
+
+    ##### COMMON ASSET NODE INTERFACE
+
+    @property
+    @cached_method
+    def group_name(self) -> Optional[str]:
+        return self._priority_node.group_name
```

Comment:
`cached_method` is about 20x more expensive than bare property access, so we need to make sure that calculating it is worth. This very well may be slower.

Reply:
Good point, removed